src/windows.rs

```diff
@@ -33,17 +39,33 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
                 BCRYPT_USE_SYSTEM_PREFERRED_RNG,
             )
         };
-        // NTSTATUS codes use the two highest bits for severity status.
-        if ret >> 30 == 0b11 {
-            // We zeroize the highest bit, so the error code will reside
-            // inside the range designated for OS codes.
-            let code = ret ^ (1 << 31);
-            // SAFETY: the second highest bit is always equal to one,
-            // so it's impossible to get zero. Unfortunately the type
-            // system does not have a way to express this yet.
-            let code = unsafe { NonZeroU32::new_unchecked(code) };
-            return Err(Error::from(code));
+        if ret > 0 {
+            // Failed. Try RtlGenRandom as a fallback.
+            let fallback_ret = fallback_rng(chunk);
+            // NTSTATUS codes use the two highest bits for severity status.
+            if fallback_ret > 0 {
+                // We zeroize the highest bit, so the error code will reside
+                // inside the range designated for OS codes.
+                let code = ret ^ (1 << 31);
+                // SAFETY: the second highest bit is always equal to one,
+                // so it's impossible to get zero. Unfortunately the type
+                // system does not have a way to express this yet.
+                let code = unsafe { NonZeroU32::new_unchecked(code) };
+                return Err(Error::from(code));
+            }
         }
     }
     Ok(())
 }
+
+#[cfg(not(target_vendor = "uwp"))]
+#[inline(never)]
+fn fallback_rng(chunk: &mut [MaybeUninit<u8>]) -> u8 {
+    unsafe { RtlGenRandom(chunk.as_mut_ptr() as *mut u8, chunk.len() as u32) }
+}
+
+#[cfg(target_vendor = "uwp")]
+#[inline(never)]
+fn fallback_rng(_chunk: &mut [MaybeUninit<u8>]) -> u8 {
```

Comment:
The return type does not seem to match the return value here? `RtlGenRandom` return no more information than TRUE or FALSE (respectively a non-zero or a zero `u8` value), so I think `fallback_rng` could return `bool` or `Maybe<bool>` depending on whether or not we want to be able to distinguish "`RtlGenRandom` is unavailable" from "`RtlGenRandom` failed". Given that at the moment we do not need to be able to distinguish since the error we propagate will be the one from `BCryptGenRandom`, maybe `bool` is enough here.